lib/open_food_network/order_cycle_permissions.rb

```diff
@@ -53,20 +74,38 @@ def visible_enterprises
           managed_active_ids = managed_participating_enterprises.pluck(:id)
 
           # TODO: Remove this when all P-OC are sorted out
-          # Any hubs that currently have outgoing exchanges distributing variants of producers I manage
-          variants = Spree::Variant.joins(:product).where('spree_products.supplier_id IN (?)', managed_enterprises.is_primary_producer.select("enterprises.id"))
+          # Hubs that currently have outgoing exchanges distributing variants of producers I manage
+          variants = Spree::Variant.
+            joins(:product).
+            where("spree_products.supplier_id IN (?)",
+                  managed_enterprises.is_primary_producer.select("enterprises.id"))
+
           active_exchanges = @order_cycle.
             exchanges.outgoing.with_any_variant(variants.select("spree_variants.id"))
+
           hubs_active_ids = active_exchanges.map(&:receiver_id)
 
           # TODO: Remove this when all P-OC are sorted out
           # Any producers of variants that hubs I manage are currently distributing in this OC
-          variant_ids = Spree::Variant.joins(:exchanges).where("exchanges.receiver_id IN (?) AND exchanges.order_cycle_id = (?) AND exchanges.incoming = 'f'", managed_participating_hubs.select("enterprises.id"), @order_cycle).pluck(:id).uniq
-          product_ids = Spree::Product.joins(:variants_including_master).where("spree_variants.id IN (?)", variant_ids).pluck(:id).uniq
-          producers_active_ids = Enterprise.joins(:supplied_products).where("spree_products.id IN (?)", product_ids).pluck(:id).uniq
+          variant_ids = Spree::Variant.joins(:exchanges).
+            where(
+              "exchanges.receiver_id IN (?)
+                AND exchanges.order_cycle_id = (?)
+                AND exchanges.incoming = 'f'",
+              managed_participating_hubs.select("enterprises.id"),
+              @order_cycle
+            ).pluck(:id).uniq
```

Comment:
These changes make me see all these little things that could be optimised. But it's not the scope of this PR and we should probably only optimise if it's a problem.

Reply:
yes! this class is complex and there's much work we could do to make this more simple and for sure more optimised. I tried to keep it as simple as possible.
getting your brain around each of these queries takes time.